sql/src/main/java/io/crate/execution/dml/upsert/GeneratedColsFromRawInsertSource.java

```diff
@@ -78,4 +92,24 @@ public BytesReference generateSource(Object[] values) throws IOException {
         return BytesReference.bytes(XContentFactory.jsonBuilder().map(source));
     }
 
+    private Map<Reference, Object> buildDefaults(List<Reference> defaults,
+                                                 EvaluatingNormalizer normalizer,
+                                                 TransactionContext txnCtx) {
+        HashMap<Reference, Object> m = new HashMap<>();
+        for (Reference ref : defaults) {
+            Symbol defaultExpression = normalizer.normalize(ref.defaultExpression(), txnCtx);
```

Comment:
Would it make sense to use `SymbolEvaluator` here instead, given that we want to evaluate it to a value? 

From its javadocs:

> This should be preferred over {@link EvaluatingNormalizer} if it's required to fully evaluate the tree

Reply:
right, my bad.